File: Actividad5/Ejercicio1/CarbonFootprintProject/src/service/FileManager.java
package service;

import java.io.FileWriter;
import java.io.IOException;

public class FileManager {

    public static void saveToFile(String data) {
        try (FileWriter writer = new FileWriter("data/output.txt", true)) {
            writer.write(data + "\n");
        } catch (IOException e) {
            e.printStackTrace();
        }
    }

    public static void clearFile() {
        try (FileWriter writer = new FileWriter("data/output.txt")) {
            writer.write(""); // deja el archivo vacío
        } catch (IOException e) {
            e.printStackTrace();
        }
    }
}

File: Actividad6/Ejercicio 1/CarbonFootprintPy/service/FileManager.py
import os

class FileManager:
    FILE_PATH = "data/output.txt"

    @staticmethod
    def save_to_file(data: str):

        os.makedirs(os.path.dirname(FileManager.FILE_PATH), exist_ok=True)
        

        with open(FileManager.FILE_PATH, "a", encoding="utf-8") as writer:
            writer.write(data + "\n")

    @staticmethod
    def clear_file():

        os.makedirs(os.path.dirname(FileManager.FILE_PATH), exist_ok=True)
        
        # 'w' sobreescribe el archivo, dejándolo vacío
        with open(FileManager.FILE_PATH, "w", encoding="utf-8") as writer:
            writer.write("")
            